Building static libraries on linux failes

I'm using this:
```
#!/bin/bash
set -e
rm -rf build.paho
mkdir build.paho
cd build.paho
echo "travis build dir $TRAVIS_BUILD_DIR pwd $PWD"
#cmake -DCMAKE_BUILD_TYPE=Debug -DPAHO_WITH_SSL=TRUE -DPAHO_BUILD_DOCUMENTATION=FALSE -DPAHO_BUILD_SAMPLES=TRUE ..
cmake -DCMAKE_BUILD_TYPE=Debug -DPAHO_BUILD_STATIC=TRUE -DPAHO_WITH_SSL=TRUE -DPAHO_BUILD_DOCUMENTATION=FALSE -DPAHO_BUILD_SAMPLES=TRUE ..
make
...

```
failes with:

CMake Error at src/CMakeLists.txt:95 (ADD_LIBRARY):
  add_library cannot create target "paho-mqtt3c" because another target with
  the same name already exists.  The existing target is a shared library
  created in source directory "/10_data/55_git/paho.mqtt.c/src".  See
  documentation for policy CMP0002 for more details.


CMake Error at src/CMakeLists.txt:96 (ADD_LIBRARY):
  add_library cannot create target "paho-mqtt3a" because another target with
  the same name already exists.  The existing target is a shared library
  created in source directory "/10_data/55_git/paho.mqtt.c/src".  See
  documentation for policy CMP0002 for more details.


CMake Error at src/CMakeLists.txt:101 (INSTALL):
  install TARGETS given no LIBRARY DESTINATION for shared library target
  "paho-mqtt3c".
...

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -92,13 +92,15 @@ INSTALL(TARGETS MQTTVersion
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 
 IF (PAHO_BUILD_STATIC)
-    ADD_LIBRARY(paho-mqtt3c STATIC $<TARGET_OBJECTS:common_obj> MQTTClient.c)
-    ADD_LIBRARY(paho-mqtt3a STATIC $<TARGET_OBJECTS:common_obj> MQTTAsync.c)
+    ADD_LIBRARY(paho-mqtt3c-static STATIC $<TARGET_OBJECTS:common_obj> MQTTClient.c)
+    ADD_LIBRARY(paho-mqtt3a-static STATIC $<TARGET_OBJECTS:common_obj> MQTTAsync.c)
 
-    TARGET_LINK_LIBRARIES(paho-mqtt3c ${LIBS_SYSTEM})
-    TARGET_LINK_LIBRARIES(paho-mqtt3a ${LIBS_SYSTEM})
+    TARGET_LINK_LIBRARIES(paho-mqtt3c-static ${LIBS_SYSTEM})
+    TARGET_LINK_LIBRARIES(paho-mqtt3a-static ${LIBS_SYSTEM})
+    SET_TARGET_PROPERTIES(paho-mqtt3c-static PROPERTIES OUTPUT_NAME paho-mqtt3c)
+    SET_TARGET_PROPERTIES(paho-mqtt3a-static PROPERTIES OUTPUT_NAME paho-mqtt3a)
 
-    INSTALL(TARGETS paho-mqtt3c paho-mqtt3a
+    INSTALL(TARGETS paho-mqtt3c-static paho-mqtt3a-static
         ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 ENDIF()
 
@@ -162,18 +164,20 @@ IF (PAHO_WITH_SSL)
         RUNTIME DESTINATION  ${CMAKE_INSTALL_BINDIR})
 
     IF (PAHO_BUILD_STATIC)
-        ADD_LIBRARY(paho-mqtt3cs STATIC $<TARGET_OBJECTS:common_ssl_obj> MQTTClient.c SSLSocket.c)
-        ADD_LIBRARY(paho-mqtt3as STATIC $<TARGET_OBJECTS:common_ssl_obj> MQTTAsync.c SSLSocket.c)
+        ADD_LIBRARY(paho-mqtt3cs-static STATIC $<TARGET_OBJECTS:common_ssl_obj> MQTTClient.c SSLSocket.c)
+        ADD_LIBRARY(paho-mqtt3as-static STATIC $<TARGET_OBJECTS:common_ssl_obj> MQTTAsync.c SSLSocket.c)
 
-        TARGET_LINK_LIBRARIES(paho-mqtt3cs ${OPENSSL_LIBRARIES} ${LIBS_SYSTEM})
-        TARGET_LINK_LIBRARIES(paho-mqtt3as ${OPENSSL_LIBRARIES} ${LIBS_SYSTEM})
+        TARGET_LINK_LIBRARIES(paho-mqtt3cs-static ${OPENSSL_LIBRARIES} ${LIBS_SYSTEM})
+        TARGET_LINK_LIBRARIES(paho-mqtt3as-static ${OPENSSL_LIBRARIES} ${LIBS_SYSTEM})
         SET_TARGET_PROPERTIES(
-        paho-mqtt3cs paho-mqtt3as PROPERTIES
+        paho-mqtt3cs-static paho-mqtt3as-static PROPERTIES
         VERSION ${CLIENT_VERSION}
         SOVERSION ${PAHO_VERSION_MAJOR}
         COMPILE_DEFINITIONS "OPENSSL=1")
+        SET_TARGET_PROPERTIES(paho-mqtt3cs-static PROPERTIES OUTPUT_NAME paho-mqtt3cs)
+        SET_TARGET_PROPERTIES(paho-mqtt3as-static PROPERTIES OUTPUT_NAME paho-mqtt3as)
 
-        INSTALL(TARGETS paho-mqtt3cs paho-mqtt3as
+        INSTALL(TARGETS paho-mqtt3cs-static paho-mqtt3as-static
             ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
     ENDIF()
 ENDIF()

